app/plugin.go

```diff
@@ -400,6 +402,96 @@ func (a *App) GetPlugins() (*model.PluginsResponse, *model.AppError) {
 	return resp, nil
 }
 
+// GetMarketplacePlugins returns a list of plugins from the marketplace-server,
+// and plugins that are installed locally.
+func (a *App) GetMarketplacePlugins(filter *model.MarketplacePluginFilter) ([]*model.MarketplacePlugin, *model.AppError) {
+	var result []*model.MarketplacePlugin
+	pluginSet := map[string]bool{}
+	pluginsEnvironment := a.GetPluginsEnvironment()
+	if pluginsEnvironment == nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.config.app_error", nil, "", http.StatusInternalServerError)
+	}
+
+	marketplaceClient, err := marketplace.NewClient(
+		*a.Config().PluginSettings.MarketplaceUrl,
+		a.HTTPService,
+	)
+	if err != nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.marketplace_client.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	marketplacePlugins, err := marketplaceClient.GetPlugins(filter)
+	if err != nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.marketplace_plugins.app_error", nil, err.Error(), http.StatusInternalServerError)
```

Comment:
Is there any chance we could return a more specific AppError like `app.plugin.marketplace_plugins.connection_error` when it cannot connect to the service? It is currently returning this generic `app.plugin.marketplace_plugins.app_error` for all of these errors and it would be best if I can distinguish in the client these errors so the appropriate message can be shown.

Reply:
@marianunez 
Absolutely. Are you referring to the status code? or just the detailed error string?

Reply:
I was actually referring to the error id (`app.plugin.marketplace_plugins.app_error`) to a more detailed one and I can use that as key to get the appropriate UI message.

Reply:
@marianunez 	actually looks like i'm only returning `app.plugin.marketplace_plugins.app_error` when i get an error back from the client when getting the plugins, shouldn't that suffice? I can update the string to say something connection related.